subcommands/keys/tuf_updates_apply.go

```diff
@@ -0,0 +1,58 @@
+package keys
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+)
+
+func init() {
+	applyCmd := &cobra.Command{
+		Use:   "apply",
+		Short: "Apply staged TUF root updates for the Factory",
+		Run:   doTufUpdatesApply,
+	}
+	applyCmd.Flags().StringP("txid", "x", "", "TUF root updates transaction ID.")
+	_ = applyCmd.MarkFlagRequired("txid")
+	tufUpdatesCmd.AddCommand(applyCmd)
+}
+
+func doTufUpdatesApply(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	txid, _ := cmd.Flags().GetString("txid")
+
+	err := api.TufRootUpdatesApply(factory, txid)
+	if err != nil {
+		msg := "Failed to apply staged TUF root updates:\n%w\n"
+		var isNonFatal bool
```

Comment:
Is connection or request timeout a fatal error? iiuc the code the timeouts will be considered as a fatal error?

Reply:
Yes, any error except the listed above is fatal; because we don't know if the updates were already partially applied or not.

The majority of 4xx codes are OK - they usually signal about a user's error, and we don't try to apply updates in those cases.

Reply:
So, if a timeout happens then a user should `cancel` and then try everything from scratch or run `apply` again?

Reply:
iiuc, in the case of timeout they can just retry `apply` no need in `cancel` and trying the overall update from scratch. My point is that `timeout` is not exactly `fatal` error.

Reply:
The most safe operation would be to run the `apply` command again.

Depending onto the phase when the timeout occured the user may or may not be able to run the `cancel` command. Most probably not.